apollo-federation/src/schema/field_set.rs

```diff
@@ -166,6 +166,38 @@ pub(crate) fn collect_target_fields_from_field_set(
     Ok(fields)
 }
 
+pub(crate) fn parse_field_value_without_validation(
+    schema: &ValidFederationSchema,
+    parent_type_name: NamedType,
+    field_value: &str,
+) -> Result<FieldSet, FederationError> {
+    // Note this parsing takes care of adding curly braces ("{" and "}") if they aren't in the
+    // string.
+    Ok(FieldSet::parse(
+        schema.schema(),
+        parent_type_name,
+        field_value,
+        "field_set.graphql",
+    )?)
+}
+
+pub(crate) fn validate_field_value(
```

Comment:
This can probably use a little note on why we need to strip check for absence of aliases.

Reply:
Addressed in [259fe72](https://github.com/apollographql/router/pull/6380/commits/259fe723715f9466bbdc6c7a9430766df73e98d7).